### PR TITLE
linter/lintmain: properly categorize packages

### PR DESCRIPTION
**Pull request by Quasilyte**
This solves issue described in
https://github.com/go-critic/go-critic/issues/715

Signed-off-by: Iskander Sharipov <quasilyte@gmail.com>

### CODE REVIEW

File: linter/lintmain/internal/check/check.go [hunk 1]
```diff
@@ -422,33 +422,62 @@ func loadPackages(cfg *packages.Config, patterns []string) ([]*packages.Package,
...
+		externalTest *packages.Package
+	}
+	packs := make(map[string]*pack)
+	internPack := func(key string) *pack {
+		if p := packs[key]; p != nil {
+			return p
+		}
+		p := &pack{}
+		packs[key] = p
+		return p
```
Comment (cristaloleg) [inline linter/lintmain/internal/check/check.go]:
```
p, ok := packs[key]
if !ok {
	p = &pack{}
	packs[key] = p
}
return p
```
wdyt ?

File: linter/lintmain/internal/check/check.go [hunk 2]
```diff
@@ -422,33 +422,62 @@ func loadPackages(cfg *packages.Config, patterns []string) ([]*packages.Package,
...
+	}
+	mustBeNil := func(pkg *packages.Package) {
+		if pkg != nil {
+			panic(fmt.Sprintf("nil assertion failed"))
```
Comment (cristaloleg) [inline linter/lintmain/internal/check/check.go]: `fmt.Sprintf` is not needed here 😉 
or `pkg.Foo` as a param is missing

**Comment by Quasilyte** [comment]
It failed on GOROOT.

File: linter/lintmain/internal/check/check.go (hunk 2 above)
Comment (Quasilyte) [reply linter/lintmain/internal/check/check.go]: Ooo-ops. Thanks for pointing that out.

**Pull request merged**